internal/backends/qt/qt_widgets/tablecolumn.rs

```diff
@@ -0,0 +1,167 @@
+        LayoutInfo { min, preferred: min, ..LayoutInfo::default() }
+    }
+
+    fn input_event_filter_before_children(
+        self: Pin<&Self>,
+        _: MouseEvent,
+        _window_adapter: &Rc<dyn WindowAdapter>,
+        _self_rc: &ItemRc,
+    ) -> InputEventFilterResult {
+        InputEventFilterResult::ForwardAndIgnore
+    }
+
+    fn input_event(
+        self: Pin<&Self>,
+        _event: MouseEvent,
+        _window_adapter: &Rc<dyn WindowAdapter>,
+        _self_rc: &i_slint_core::items::ItemRc,
+    ) -> InputEventResult {
+        InputEventResult::EventIgnored
+    }
+
+    fn key_event(
+        self: Pin<&Self>,
+        _: &KeyEvent,
+        _window_adapter: &Rc<dyn WindowAdapter>,
+        _self_rc: &ItemRc,
+    ) -> KeyEventResult {
+        KeyEventResult::EventIgnored
+    }
+
+    fn focus_event(
+        self: Pin<&Self>,
+        _: &FocusEvent,
+        _window_adapter: &Rc<dyn WindowAdapter>,
+        _self_rc: &ItemRc,
+    ) -> FocusEventResult {
+        FocusEventResult::FocusIgnored
+    }
+
+    fn_render! { this dpr size painter widget initial_state =>
+        let index: i32 = this.index();
+        let has_hover: bool = this.has_hover();
+        let item = this.item();
+        let text: qttypes::QString = item.title.as_str().into();
+        let ascending: bool = item.sort_by_ascending;
+        let descending: bool = item.sort_by_descending;
+
+        cpp!(unsafe [
+            painter as "QPainterPtr*",
+            widget as "QWidget*",
+            size as "QSize",
+            dpr as "float",
+            index as "int",
+            has_hover as "bool",
+            text as "QString",
+            initial_state as "int",
+            ascending as "bool",
+            descending as "bool"
+        ] {
+            QStyleOptionHeader option;
+            option.state |= QStyle::State(initial_state);
+            option.state |= QStyle::State_Horizontal;
+            option.rect = QRect(QPoint(), size / dpr);
+
+            option.section = index;
+
+            option.textAlignment = Qt::AlignLeft | Qt::AlignVCenter;
+           
+
+            if (ascending) {
+                option.sortIndicator = QStyleOptionHeader::SortDown;
+            } else if (descending) {
+                option.sortIndicator = QStyleOptionHeader::SortUp;
+            } else {
+                option.sortIndicator = QStyleOptionHeader::None;
+            }
+
+            if (has_hover) {
+                option.state |= QStyle::State_MouseOver;
+            }
+
+            option.text = text;
+            // CE_ItemViewItem in QCommonStyle calls setClipRect on the painter and replace the clips. So we need to cheat.
+            auto engine = (*painter)->paintEngine();
+            auto old_clip = engine->systemClip();
+            auto new_clip = old_clip & ((*painter)->clipRegion() * (*painter)->transform());
+            if (new_clip.isEmpty()) return;
+            engine->setSystemClip(new_clip);
```

Comment:
You don't need this hack, this is because of a bug in CE_ItemViewItem, but that's not what you are drawing here.

Reply:
removed